File: algorithms/Strings/Easy/SplitAStringInBalancedStrings/VSDSplitAStringInBalancedStrings.py


def balancedStringSplit( string : str ) -> int :
    check=answer=0
    for char in string:
        if char=="R":
            check+=1
        else:
            check-=1
        if check==0:
            answer+=1
    return answer

input="LLLLRRRR"
print(balancedStringSplit(input))

File: algorithms/Strings/Easy/SplitAStringInBalancedStrings/VSDSplitAStringInBalancedStrings.java
public class Main{

    static int balanceStringSplit(String s){
        int check=0,answer=0;
        for(char ch:s.toCharArray()){
            if(ch=='R')check++;
            else check--;
            if(check==0) answer++;
        }
        return answer;
    }
    public static void main(String[] args) {
        String s="RLRRLLRLRL";
        System.out.println(balanceStringSplit(s));
    }
}